app/screens/search/search.js

```diff
@@ -529,6 +529,27 @@ export default class Search extends PureComponent {
                     id: 'mobile.search.in_modifier_description',
                     defaultMessage: 'to find posts in specific channels',
                 }),
+            }, {
+                value: 'on:',
+                modifier: `on:${intl.formatMessage({id: 'mobile.search.on_modifier_title', defaultMessage: 'YYYY-MM-DD'})}`,
```

Comment:
is the date format specific to the locale? or is it always needed in this format by the server? if the later then it should not be localized, same with the other two